pandas/tests/computation/test_eval.py

```diff
@@ -1865,6 +1865,32 @@ def test_invalid_engine():
         pd.eval("x + y", local_dict={"x": 1, "y": 2}, engine="asdf")
 
 
+@td.skip_if_no_ne
+@pytest.mark.parametrize("use_numexpr", [True, False])
+def test_numexpr_option_respected(use_numexpr):
+    # GH 32556
+    from pandas.core.computation.eval import _check_engine
+
+    with pd.option_context("compute.use_numexpr", use_numexpr):
+        result = _check_engine(None)
+        if use_numexpr:
+            assert result == "numexpr"
+        else:
+            assert result == "python"
```

Comment:
logic (e.g. if/else statements) in tests aren't ideal (https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html), how about something like
```
@pytest.mark.parametrize(
    ("use_numexpr", "expected"),
    (
          (True, "numexpr"),
          (False, "python"),
    )
)
def test_numexpr_option_respected(use_numexpr, expected):
```
and then, in the body, just `assert result == expected`?

Reply:
This is good to know! Thank you for linking the resource. Added!